src/Microsoft.AspNet.Authentication.OpenIdConnect/OpenIdConnectAuthenticationHandler.cs

```diff
@@ -217,6 +218,47 @@ public OpenIdConnectAuthenticationHandler(HttpClient backchannel)
 
             message.State = Options.StateDataFormat.Protect(properties);
 
+            if (Options.UsePostAuthorizationRequests)
+            {
+                using (var buffer = new MemoryStream())
+                using (var writer = new StreamWriter(buffer))
+                {
+                    writer.WriteLine("<!doctype html>");
```

Comment:
Nit: This handcrafted html, can we put this into a string constant at least and just format string the action and input block in a which you generate as well?  

Something like this so you can just put all the html in one literal string constant block formatted how you like without all the writelines

```
var inputs = new StringBuilder();
foreach (var parameter in message.Parameters) {
   inputs.Add(string.Format(InputTagFormat, name, value);
}
writer.Write(string.Format(HtmlFormat, messageAction, inputs));
```